holoviews/core/options.py

```diff
@@ -474,21 +485,26 @@ def max_cycles(self, num):
         """
         kwargs = {kw: (arg[num] if isinstance(arg, Cycle) else arg)
                   for kw, arg in self.kwargs.items()}
-        return self(**kwargs)
```

Comment:
I think this docstring should say this is about discretizing palettes. I also think the method name is very confusing but that would be an API change so it would not be something to fix in this PR...